Plugin.pm

```diff
@@ -1005,32 +1012,74 @@ sub QobuzGetTracks {
 				# Qobuz sometimes would f... up work names, randomly putting whitespace etc. in names - ignore them
 				my $workId = Slim::Utils::Text::matchCase(Slim::Utils::Text::ignorePunct($work));
 				$workId =~ s/\s//g;
+				my $displayWorkId = Slim::Utils::Text::matchCase(Slim::Utils::Text::ignorePunct($formattedTrack->{displayWork}));
+				$displayWorkId =~ s/\s//g;
+
+				# Unique work identifier, used to keep tracks together even if composer is missing from some, but on the other hand
+				# still distinguishing between works with the same name but different composer!
+				$currentComposer = $track->{composer}->{name};
+				if ( $workId eq $lastwork && (!$lastComposer || !$currentComposer || $lastComposer eq $currentComposer) ) {
+					# Stick with the previous value! ($worksWorkId = $worksWorkId;) 
+				} elsif ( $currentComposer ) {
+					$worksWorkId = $displayWorkId;
+				} else {
+					$worksWorkId = $workId;
+				}
 
-				$works->{$workId} = {   # create a new work object
-					index => $i,		# index of first track in the work
-					title => $formattedTrack->{displayWork},
-					tracks => []
-				} unless $works->{$workId};
-
-				if ($workId ne $lastwork) {  # create a new work heading
+				# Extended Work ID: will usually not change, but we need to keep non-contiguous tracks from the same work
+				# separate if the user has chosen to integrate playlists with the work titles.				
+				$worksWorkIdE = $worksWorkId;					
+				if ( $workPlaylistPos eq "integrated" && $works->{$worksWorkId} ) {
+					if ( $worksWorkId ne $lastWorksWorkId ) {
+						$discontigWorks->{$worksWorkId} = $worksWorkId . $i;
```

Comment:
Is it time to name `$i` something meaningful?...

Reply:
OK, I've changed it to `$trackNumber` and then `$workNumber` (it was used in two different loops) in the `QobuzGetTracks` subroutine.

There is also a `$i` in subroutine `_getGoodiesCLI` and I'm not certain of its meaning there - goodieNumber?

Reply:
I would often use `$i` in simple loops, where its meaning is relatively obvious. But in the above case the loop has grown considerably...